shared/plot_morpho.m

```diff
@@ -0,0 +1,275 @@
+function 	[halfwidth, peaktrough, troughpeak] = plot_morpho(cfg,data)
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% [halfwidth, peaktrough, troughpeak] = plot_morpho(cfg,data)
+% Plot trials and average of trials. If asked, do measurements on it.
+%
+% Note :
+% - If there are more than 1000 trials, only 1000 random-selected trials
+% are plotted. The average (and std) remain calculated on all the trials.
+% - Measurements may not work well if data events are unclear or artefacted
+%
+% ### Necessary input
+% data                    = raw Fieldtrip data structure epoched in trials
+% cfg.channame            = name of the channel to analyse (in data.label)
+% 
+% ### Optionnal cfg fields :
+% cfg.plotstd             = 'yes' or 'no', whether to plot std or not.
+%                           Default = 'no'.
+% cfg.toiplot             = x limits of plot. Can be 'all'. Default = 'all'.
+% cfg.saveplot            = 'yes' or 'no', whether to save and close the
+%                           plot or to output it (respectively). Default =
+%                           'no'.
+% cfg.removeoutliers      = whether to plot outlier trials (>10*std to the
+%                           mean). Average is not modified. Default = 'no'.
+% cfg.mesurehalfwidth     = 'yes' or 'no', whether to compute halfwidth.
+%                           Default = 'no'.
+% cfg.halfwidthmethod     = 'min' or 'bl' : reference for peak-amplitude
+%                           measurement if mesurehalfwidth = 'yes'. 
+%                           Default = 'bl' (baseline)
+% cfg.mesurepeaktrough    = 'yes' or 'no', whether to compute peak-trough
+%                           and trough-peak. Default = 'no'.
+% 
+% ### Necessary cfg fields if cfg.mesurehalfwidth = 'yes' or cfg.mesurepeaktrough = 'yes'
+% cfg.toiac               = active period for measurements. Can be 'all'
+% cfg.toibl               = baseline period if cfg.halfwidthmethod = 'bl'
+% 
+% ### Necessary cfg fields if cfg.saveplot = 'yes'
+% cfg.name                = name of the analysis (title of plot and of
+%                           output image file)
+% cfg.imagesavedir        = where to save the image.
+% cfg.prefix              = prefix attached to the name of the saved-image
+%
+% ### OUTPUT
+% halfwidth               = mesured halfwidth value, in seconds, or [].
+% peaktrough              = mesured peaktrough value, in seconds, or [].
+% troughpeak              = mesured troughpeak value, in seconds, or [].
+%
+% Paul Baudin (paul.baudin@live.fr)
```

Comment:
To clarify: Here and in the other places where you present yourself as the author of the code; as long as you are using (some) code from someone else, you should credit them. Here and elsewhere it would be more appropriate to say something like "Code by Stephen Whitmarsh & Paul Baudin"". Also note that the EpiCode is shared under a specific (GNU) license (see license file) in my name, which is a very specific kind of copyright ("copyleft") that prohibits claims of ownership by anyone (including me).